Update Github action workflows

tylervz opened the pull request:
Update the GitHub actions workflows to use the latest version of multiple actions, including `actions/checkout`,  `actions/setup-java`, `actions/cache`, and `release-drafter/release-drafter`:

https://github.com/actions/checkout/tags
https://github.com/actions/setup-java/tags
https://github.com/actions/cache?tab=readme-ov-file#v4
https://github.com/release-drafter/release-drafter/tags

tylervz committed:
chore: update GH actions workflows

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
We are trying to switch to `liberica` across the board.

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
           distribution: temurin
           java-version: 17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Use v4

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
```
We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew build
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
`DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
Use `liberica` instead of `temurin`

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
           distribution: temurin
           java-version: '17'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Use v4

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          build-scan-publish: true
```
We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -10,48 +10,58 @@ jobs:
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew -Dorg.gradle.internal.publish.checksums.insecure=true publish
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
`DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

matrei commented on .github/workflows/release.yml:
```diff
@@ -13,11 +13,11 @@ jobs:
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
```
Use `liberica` instead of `temurin`

matrei commented on .github/workflows/release.yml:
```diff
@@ -35,12 +35,18 @@ jobs:
         id: release_version
         run: echo "value=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Use v4

matrei commented on .github/workflows/release.yml:
```diff
@@ -35,12 +35,18 @@ jobs:
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
```
We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

matrei commented on .github/workflows/release.yml:
```diff
@@ -35,12 +35,18 @@ jobs:
-          arguments: assemble
+        run: ./gradlew assemble
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
`DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

matrei commented on .github/workflows/release.yml:
```diff
@@ -57,7 +63,10 @@ jobs:
+          publishToSonatype 
+          closeSonatypeStagingRepository
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
`DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

matrei commented on .github/workflows/release.yml:
```diff
@@ -84,19 +88,23 @@ jobs:
           token: ${{ secrets.GH_TOKEN }}
           ref: v${{ needs.publish.outputs.release_version }}
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
```
Use v4

matrei commented on .github/workflows/release.yml:
```diff
@@ -84,19 +88,23 @@ jobs:
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          build-scan-publish: true
```
We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.

matrei commented on .github/workflows/release.yml:
```diff
@@ -84,19 +88,23 @@ jobs:
+          ./gradlew findSonatypeStagingRepository
+          releaseSonatypeStagingRepository 
         env:
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
`DEVELOCITY_ACCESS_KEY` is not necessary here. A short-lived access key It is provided by `setup-gradle` thanks to `with: develocity-access-key: ...`

tylervz commented:
@matrei I updated all the `gradle/actions/setup-gradle` to v4 and declared them with
```
- name: Setup Gradle
        uses: gradle/actions/setup-gradle@v4
        with:
          develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
And then I removed the `DEVELOCITY_ACCESS_KEY` environment variable declarations.

matrei commented:
@tylervz  Thanks!, looking good :star:
